pkg/datarepair/queue/queue.go

```diff
@@ -0,0 +1,82 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package queue
+
+import (
+	"encoding/binary"
+	"sync"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/storage"
+	"storj.io/storj/storage/redis"
+)
+
+//RepairQueue is the interface for the data repair queue
+type RepairQueue interface {
+	Add(qi *pb.InjuredSegment) (storage.Key, error)
+	Remove(date storage.Key) error
+	GetNext() (storage.Key, pb.InjuredSegment, error)
+	GetSize() (int, error)
```

Comment:
Is GetSize necessary here. Given that this is a concurrent queue it will start changing everytime you call it. For statistics the queue itself can handle it.